spec/models/presentation_spec.rb

```diff
@@ -1,5 +1,87 @@
+# spec/models/presentation_spec.rb
 require 'rails_helper'
 
 RSpec.describe Presentation, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  let(:person) { FactoryGirl.create(:person) }
+  let(:event)  { FactoryGirl.create(:event) }
+
+  describe 'validate model' do
+    subject { FactoryGirl.create(:presentation) }
+
+    it { is_expected.to validate_presence_of :topic }
+    it { is_expected.to validate_presence_of :duration }
+
+    it { is_expected.to belong_to :person }
+    it { is_expected.to belong_to :event }
+
+    it { is_expected.to be_valid }
+
+    describe 'person and presenter validation' do
+      subject { described_class.new(presentation) }
+
+      context 'with presenter, no person' do
+        let(:presentation) do
+          FactoryGirl.attributes_for(
+            :presentation,
+            event_id: event.id,
+            presenter: 'test'
+          )
+        end
+
+        it { is_expected.to be_valid }
+        it { is_expected.to have_attributes(person: nil, presenter: 'test') }
+      end
+
+      context 'with person, no presenter' do
+        let(:presentation) do
+          FactoryGirl.attributes_for(
+            :presentation,
+            event_id: event.id,
+            person_id: person.id
+          )
+        end
+
+        it { is_expected.to have_attributes(person: person, presenter: nil) }
+        it { is_expected.to be_valid }
+      end
+
+      context 'with presenter, and person' do
+        let(:presentation) do
+          FactoryGirl.attributes_for(
+            :presentation,
+            event_id: event.id,
+            presenter: 'test',
+            person_id: person.id
+          )
+        end
+
+        it { is_expected.to have_attributes(person: person, presenter: 'test') }
+        it { is_expected.to_not be_valid }
+      end
+
+      context 'without presenter, or person' do
+        let(:presentation) do
+          FactoryGirl.attributes_for(:presentation, event_id: event.id)
+        end
+
+        it { is_expected.to_not be_valid }
+      end
+    end
+  end
+
+  describe 'presenter_name' do
+    context 'when presenter is not present' do
+      subject { FactoryGirl.create(:presentation) }
+
+      it { is_expected.to have_attributes(presenter_name: 'John') }
```

Comment:
its testing the dynamic property when person is set, and presenter is nil. In the Factories, this is specified as 'John'